les/handler.go

```diff
@@ -853,6 +866,7 @@ func (pm *ProtocolManager) handleMsg(p *peer) error {
 
 						if number = rawdb.ReadHeaderNumber(pm.chainDb, request.BHash); number == nil {
 							p.Log().Warn("Failed to retrieve block num for proof", "hash", request.BHash)
+							atomic.AddUint32(&p.invalidCount, 1)
```

Comment:
If the request contain N internal requests for the same missing hash, then this path won't trigger. Might be better to base it off #19701 , and add the counter here: https://github.com/ethereum/go-ethereum/pull/19701/files#diff-d1e187d1b095dcfc1cc773abfd8c4066R873